mkdocs.yml

```diff
@@ -38,22 +38,37 @@ nav:
   - ./tutorial/udl_file.md
   - ./tutorial/Rust_scaffolding.md
   - ./tutorial/foreign_language_bindings.md
-- 'The UDL file':
-  - ./udl_file_spec.md
-  - ./udl/namespace.md
-  - ./udl/builtin_types.md
-  -  ./udl/enumerations.md
-  -  ./udl/structs.md
+- 'UniFFI type model':
+  - ./types/namespace.md
+  - ./types/builtin_types.md
+  - ./types/enumerations.md
+  - ./types/structs.md
   - 'Functions':
-    - ./udl/functions.md
-    - ./udl/errors.md
+    - ./types/functions.md
+    - ./types/errors.md
   - 'Interfaces and Objects':
+    - ./types/interfaces.md
+    - ./types/callback_interfaces.md
+  - ./types/remote_ext_types.md
+  - ./types/custom_types.md
+- Specifying the interface:
+  - ./specifying.md
+  - 'UDL Files':
+    - ./udl/index.md
+    - ./udl/enumerations.md
+    - ./udl/errors.md
+    - ./udl/functions.md
     - ./udl/interfaces.md
-    - ./udl/callback_interfaces.md
-  - ./udl/remote_ext_types.md
-  - ./udl/custom_types.md
-  - ./udl/docstrings.md
-- ./proc_macro/index.md
+    - ./udl/records.md
+    - ./udl/docstrings.md
+  - 'Proc macros':
+    - ./proc_macro/index.md
+    - ./proc_macro/enumerations.md
+    - ./proc_macro/errors.md
+    - ./proc_macro/functions.md
+    - ./proc_macro/interfaces.md
+    - ./proc_macro/records.md
+    - ./proc_macro/docstrings.md
```

Comment:
Maybe we don't all of these TOC entries.  Like @joe-p says, if I wanted to learn about records and I was using proc-macros, I'm still probably better off starting an `types/records.md`.

Instead, there could just one top-level proc-macro TOC entry, which would contain the overview on proc-macros and a link to the docstrings info.

Reply:
That's where I started too, but I think this ends up working better in the rendered versions given they are default collapsed. I agree there's work needed here though!